src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs

```diff
@@ -9,17 +16,256 @@
     /// </summary>
     public class RabbitMQTransport : TransportDefinition
     {
+        static readonly TransportTransactionMode[] supportedTransactionModes = 
+        {
+            TransportTransactionMode.None,
+            TransportTransactionMode.ReceiveOnly
+        };
+
+        TimeSpan timeToWaitBeforeTriggeringCircuitBreaker = TimeSpan.FromMinutes(2);
+        Func<BasicDeliverEventArgs, string> messageIdStrategy = MessageConverter.DefaultMessageIdStrategy;
+        IRoutingTopology routingTopology = new ConventionalRoutingTopology(true);
+        Func<int, int> prefetchCountCalculation = maxConcurrency => 3 * maxConcurrency;
+        string host;
+        TimeSpan heartbeatInterval = TimeSpan.FromMinutes(1);
+        TimeSpan networkRecoveryInterval = TimeSpan.FromSeconds(10);
+
+        /// <summary>
+        /// The host to connect to.
+        /// </summary>
+        public string Host
+        {
+            get => host;
+            set
+            {
+                Guard.AgainstNullAndEmpty("value", value);
+                host = value;
+            }
+        }
+
+        /// <summary>
+        /// The port to connect to.
+        ///
+        /// If not specified, the default port will be used (5672 if not encrypted and 5671 if using TLS)
+        /// </summary>
+        public int? Port { get; set; }
+
+        /// <summary>
+        /// The vhost to connect to.
+        /// </summary>
+        public string VHost { get; set; } = "/";
+
+        /// <summary>
+        /// The user name to pass to the broker for authentication.
+        /// </summary>
+        public string UserName { get; set; } = "guest";
+
+        /// <summary>
+        /// The password to pass to the broker for authentication.
+        /// </summary>
+        public string Password { get; set; } = "guest";
+
+        /// <summary>
+        /// The routing topology to use. If not set the conventional routing topology will be used <seealso cref="ConventionalRoutingTopology"/>.
+        /// </summary>
+        public IRoutingTopology RoutingTopology
```

Comment:
Ok, yeah this is what I afraid of when I saw that the topology classes had been made public.. This is a strictly worse configuration experience compared to what we had before. This puts the onus on the caller to have to create a topology class in all instances of using this API.

In the past that would only have been required when they had created their own custom topology implementation.

Reply:
They only need to do it if they chose to select a non-default topology. In this version the conventional topology is made the default. Since very few users use the direct one, I don't this is going to be a major issue for the majority.